cmd/fs-v1-background-append.go

```diff
@@ -186,29 +188,27 @@ func (b *backgroundAppend) appendParts(disk StorageAPI, bucket, object, uploadID
 
 // Appends the "part" to the append-file inside "tmp/" that finally gets moved to the actual location
 // upon complete-multipart-upload.
-func appendPart(disk StorageAPI, bucket, object, uploadID string, part objectPartInfo, buf []byte) error {
-	partPath := pathJoin(bucket, object, uploadID, part.Name)
-
-	offset := int64(0)
-	totalLeft := part.Size
-	for totalLeft > 0 {
-		curLeft := int64(readSizeV1)
-		if totalLeft < readSizeV1 {
-			curLeft = totalLeft
-		}
-		n, err := disk.ReadFile(minioMetaMultipartBucket, partPath, offset, buf[:curLeft])
-		if err != nil {
-			// Check for EOF/ErrUnexpectedEOF not needed as it should never happen as we know
-			// the exact size of the file and hence know the size of buf[]
-			// EOF/ErrUnexpectedEOF indicates that the length of file was shorter than part.Size and
-			// hence considered as an error condition.
-			return err
-		}
-		if err = disk.AppendFile(minioMetaTmpBucket, uploadID, buf[:n]); err != nil {
-			return err
+func (fs fsObjects) appendPart(bucket, object, uploadID string, part objectPartInfo, buf []byte) error {
+	partPath := pathJoin(fs.fsPath, minioMetaMultipartBucket, bucket, object, uploadID, part.Name)
+
+	file, err := os.OpenFile(partPath, os.O_RDONLY, 0666)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return errPartsMissing
 		}
-		offset += n
-		totalLeft -= n
+		return err
+	}
+	defer file.Close()
+
+	tmpObjPath := pathJoin(fs.fsPath, minioMetaTmpBucket, fs.fsUUID, uploadID)
+	// No need to hold a lock, this is a unique file and will be only written
```

Comment:
This is not guaranteed. The user must configure such that all multipart requests corresponding uploadID arrive on the same minio server instance.
e.g,
server1                                                                
- newMultipartUpload -> upload1                       
- putObjectPart(upload1, part1, part2, part4)
- completeMultipart(upload1, ...)

server2
- putObjectPart(upload1, part3, part4)
- completeMultipart(upload1, ...)

The background append process in server1 and server2 would be competing to write on the temporary file, which is not unique intrinsically and needs to be configured at load-balancer layer.